pkg/ruler/ruler.go

```diff
@@ -364,28 +365,23 @@ func (r *Ruler) getOrCreateNotifier(userID string) (*notifier.Manager, error) {
 func (r *Ruler) Evaluate(userID string, item *workItem) {
 	ctx := user.InjectOrgID(context.Background(), userID)
 	logger := util.WithContext(ctx, util.Logger)
-	level.Debug(logger).Log("msg", "evaluating rules...", "num_rules", len(item.rules))
+	level.Debug(logger).Log("msg", "evaluating rules...", "num_rules", len(item.group.Rules()))
 	ctx, cancelTimeout := context.WithTimeout(ctx, r.groupTimeout)
 	instrument.CollectedRequest(ctx, "Evaluate", evalDuration, nil, func(ctx native_ctx.Context) error {
 		if span := opentracing.SpanFromContext(ctx); span != nil {
 			span.SetTag("instance", userID)
 			span.SetTag("groupName", item.groupName)
 		}
-		g, err := r.newGroup(ctx, userID, item)
-		if err != nil {
-			level.Error(logger).Log("msg", "failed to create rule group", "err", err)
-			return err
-		}
-		g.Eval(ctx, time.Now())
+		item.group.Eval(ctx, time.Now())
 		return nil
 	})
 	if err := ctx.Err(); err == nil {
 		cancelTimeout() // release resources
 	} else {
-		level.Warn(util.Logger).Log("msg", "context error", "error", err)
+		level.Warn(util.Logger).Log("msg", "context error", "user_id", userID, "error", err)
```

Comment:
I think if you change `util.Logger` to `logger` you will get the user ID automagically.